File: auditHub/src/main/java/com/thy/audithub/dto/FilterRequest.java
package com.thy.audithub.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.LocalDate;
import java.util.List;

@Getter
@Setter
public class FilterRequest {

    @NotBlank(message = "Müdürlük seçimi zorunludur.")
    private String mudurluk;

    @NotEmpty(message = "En az bir issue type seçilmelidir.")
    private List<String> issueTypes;

    @NotNull(message = "Oluşturulma başlangıç tarihi boş olamaz.")
    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
    private LocalDate createdStartDate;

    @NotEmpty(message = "En az bir status seçilmelidir.")
    private List<String> statuses;

    @NotNull(message = "Durum değişim başlangıç tarihi boş olamaz.")
    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
    private LocalDate statusChangedStartDate;

    @NotNull(message = "Durum değişim bitiş tarihi boş olamaz.")
    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
    private LocalDate statusChangedEndDate;

    public boolean isDateRangeValid() {
        if (statusChangedStartDate == null || statusChangedEndDate == null) {
            return true;
        }
        return !statusChangedEndDate.isBefore(statusChangedStartDate);
    }

    public boolean hasCustomJql() {
        if (customJql == null || customJql.isBlank()) return false;
        if (customJql.contains("Tüm alanları doldurunca")) return false;
        return true;
    }

    private String customJql;
}

File: Models/FilterRequest.cs
using System.ComponentModel.DataAnnotations;

namespace AuditHub.Models;

public class FilterRequest
{
    [Required(ErrorMessage = "Jira token gereklidir.")]
    public string JiraToken { get; set; } = "";

    [Required(ErrorMessage = "Müdürlük seçimi gereklidir.")]
    public string Mudurluk { get; set; } = "";

    public List<string> IssueTypes { get; set; } = [];

    [Required(ErrorMessage = "Created başlangıç tarihi gereklidir.")]
    public DateOnly CreatedStartDate { get; set; }

    public List<string> Statuses { get; set; } = [];

    [Required(ErrorMessage = "Status tarihi gereklidir.")]
    public DateOnly StatusChangedStartDate { get; set; }

    [Required(ErrorMessage = "Status bitiş tarihi gereklidir.")]
    public DateOnly StatusChangedEndDate { get; set; }

    public string? CustomJql { get; set; }

    public bool IsDateRangeValid() => StatusChangedEndDate >= StatusChangedStartDate;

    public bool HasCustomJql() => !string.IsNullOrWhiteSpace(CustomJql);
}
